onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc

```diff
@@ -114,6 +117,11 @@ void CreateCoreMLWeight(CoreML::Specification::WeightParams& weight, gsl::span<c
   weight.mutable_floatvalue()->Assign(data.begin(), data.end());
 }
 
+void CreateCoreMLWeight(CoreML::Specification::WeightParams& weight, gsl::span<const MLFloat16> data) {
```

Comment:
Do we need CreateCoreMLWeightConvertingDataToFloats to also support converting to fp16? Maybe not until we add fp16 support to operators that have initializers as input.

Reply:
Already support `CreateCoreMLWeightConvertingDataToFloat16s `